pkg/proxy/endpoints.go

```diff
@@ -121,7 +121,7 @@ func (info *BaseEndpointInfo) Port() (int, error) {
 func (info *BaseEndpointInfo) Equal(other Endpoint) bool {
 	return info.String() == other.String() &&
 		info.GetIsLocal() == other.GetIsLocal() &&
-		info.IsReady() == other.IsReady()
+		info.IsServing() == other.IsServing()
```

Comment:
This feels wrong. Surely an `Endpoint` that is terminating is not "Equal" to one that is not terminating?

It seems that `Endpoint.Equal` is actually only used in `detectStaleConntrackEntries`... I think rather than having an alleged "Equal" function whose semantics have to be fine-tuned to do what `detectStaleConntrackEntries` wants, we should just remove the function from the interface (and also iptables's redefinition of it for its `endpointsInfo`), and just have `detectStaleConntrackEntries` check the specific thing that it cares about; for each old endpoint which is Serving, is there a corresponding new endpoint which is still Serving. ("Corresponding" meaning "`oldEP.String() == newEP.String()`".)